olive/__init__.py

```diff
@@ -16,6 +16,22 @@
 
 __version__ = "0.10.0.dev0"
 
+try:
+    import onnxruntime as ort
+
+    # pylint: disable=protected-access
+    ort._get_available_providers = ort.get_available_providers
+
+    def get_available_providers_winml():
+        # pylint: disable=protected-access
+        providers = ort._get_available_providers()
+        extra_providers = {ep_device.ep_name for ep_device in ort.get_ep_devices()} - set(providers)
+        return providers + list(extra_providers)
```

Comment:
one pitfall for this: the get_available_providers has certain order, we should keep the order https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/get_execution_providers.cc
I think we had better filter from get_all_providers https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/onnxruntime_pybind_state.cc#L1618